geomstats/geometry/stiefel.py

```diff
@@ -64,22 +65,19 @@ def belongs(self, point, tolerance=TOLERANCE):
             Array of booleans evaluating if the corresponding points
             belong to the Stiefel manifold.
         """
-        point = gs.to_ndarray(point, to_ndim=3)
         n_points, n, p = point.shape
 
         if (n, p) != (self.n, self.p):
-            return gs.array([[False]] * n_points)
+            return gs.array([False] * n_points)
 
-        point_transpose = gs.transpose(point, axes=(0, 2, 1))
-        identity = gs.to_ndarray(gs.eye(p), to_ndim=3)
-        identity = gs.tile(identity, (n_points, 1, 1))
-        diff = gs.einsum('nij,njk->nik', point_transpose, point) - identity
+        point_transpose = gs.swapaxes(point, axis1=-1, axis2=-2)
```

Comment:
why not use the class method Matrices.transpose here?

Reply:
done.